### PR TITLE
Share 222, 226

### PR DESCRIPTION
**Pull request by dmetzner**
- Adding **Github actions for ESLint** checks.
  - Adding JSLint to the project (npm) https://eslint.org/
  - using standard code style https://github.com/standard/standard

- **Fixing/code style** issues for all **JS** files
- Refactoring some JS files (Register,...)

- Moving Catblocks Js into catblocks directory
- **deleting** all **locally** stored (local/global)plugins and images. We are using **npm now**.
- updating package.json
- updating deprecations to newest versions (sweetalert2, textfilljs, jquery, ...)
- Removing unnecessary imports of scripts/stylesheets that were never used

- Changing the script loading process to a, **not render-blocking** way! Basically this means defining scripts only in the JS block and not in the head block. Only files that directly extend the base twig file have access to this block. Therefore some scripts which are included via twig includes have to be moved. Also, minor changes are applied to work with this flow to some twig files.

- **Reworking the Gruntfile**:
  - watch works now on the assets
  - js assets + used node_modules will be copied to public and a min.js will be created for every js file
  - dynamic theme loading (symfony liip config)
  - removing unnecessary stuff
  - splitting/describing config definitions

### Additionally
- Removing pocketphiro flavor since it is called phirocode and updating legacy tests
- Updating .gitignore for public folder
- Fixing Google Sign in on PW request page  
- Moving profile logic from twig to symfony controller

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [x] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Verify that all tests are passing, if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [x] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: assets/js/analytics/analytics.js [hunk 1]
```diff
@@ -428,51 +393,47 @@ else
...
+        const copy = new TrackingObject()
+        const toCopy = this
+
         // iterates the parameters of the TrackingObject and
         // executes functions of the parametes
         Object.keys(toCopy).forEach(function (key, index) {
           // key: the name of the object key
           // index: the ordinal position of the key within the object
-          copy[key] = ((typeof(copy[key]) === 'object') ? Object.assign({}, toCopy[key]) : toCopy[key])
+          copy[key] = ((typeof (copy[key]) === 'object') ? Object.assign({}, toCopy[key]) : toCopy[key])
```
Comment (HCrane) [inline assets/js/analytics/analytics.js:422]: This space after `typeof` seems counter intuitice but apperantly it is standard (word joke 😋 ) (https://eslint.org/docs/rules/space-unary-ops)

File: assets/js/analytics/analytics.js [hunk 2]
```diff
@@ -1208,525 +1099,499 @@ else
...
+      if (AnalyticsTracker.isSending === true) {
         gtag('js', new Date())
         // map the url
-        let href = UrlMapper.map(window.location.href, configDefault.UrlMapping)
+        const href = UrlMapper.map(window.location.href, configDefault.UrlMapping)
         gtag('config', configDefault.UaId, {
-          'anonymize_ip' : true,
-          'page_location': href,
+          anonymize_ip: true,
+          page_location: href
```
Comment (HCrane) [inline assets/js/analytics/analytics.js:1150]: Does this change have an effect?

File: assets/js/base/Main.js [hunk 3]
```diff
@@ -1,355 +1,304 @@
...
-          {
-            refrehSidebar();
-          }
-          else
-          {
-            reset();
+    })
+
+    document.addEventListener('touchmove', function (e) {
+      if (e.touches.length === 1 && (closing || opening) && !!curX) {
```
Comment (HCrane) [inline assets/js/base/Main.js:182]: Maybe I am too tired right now but `!!curX` is the same as `curX`??

File: assets/js/custom/MyProfile.js [hunk 4]
```diff
@@ -1,279 +1,270 @@
...
+    const url = stringTranslate(programName, catalogEntry)
     $.get(url, function (data) {
-      let split = data.split('\n')
-      swal({
-        title             : split[0],
-        html              : split[1] + '<br><br>' + split[2],
-        type              : 'warning',
-        showCancelButton  : true,
+      const split = data.split('\n')
+      Swal.fire({
```
Comment (HCrane) [inline assets/js/custom/MyProfile.js:119]: Was this done by eslint or via developer?

File: assets/js/custom/Notifications.js [hunk 5]
```diff
@@ -73,303 +69,261 @@ function Notification (new_notifications, old_notifications, markAsReadUrl, mark
...
+          url: self.deleteNotificationUrl + '/' + id,
+          type: 'get',
+          success: function (data) {
+            if (data.success) {
+              if ($('#new-notifications-container').children().find('#catro-notification-' + id).length) {
+                self.notifications--
+
+                self.updateNotificationAmountText()
+
+                self.updateBadgeNumber()
```
Comment (HCrane) [inline assets/js/custom/Notifications.js:287]: Remove empty lines

File: assets/js/custom/ProgramDescription.js [hunk 6]
```diff
@@ -1,93 +1,81 @@
+/* eslint-env jquery */
+/* global Routing */
+
+// eslint-disable-next-line no-unused-vars
```
Comment (HCrane) [inline assets/js/custom/ProgramDescription.js:4]: This does come somewhat often. Should we put that into gloabl eslint config?

File: assets/js/custom/Program.js [hunk 7]
```diff
@@ -1,363 +1,329 @@
...
+    if (bgDarkPopupInfo.length > 0 && data.status === 'ready') {
       bgDarkPopupInfo.remove()
     }
   }
-  
+
   self.createLinks = function () {
     $('#description').each(function () {
-      $(this).html($(this).html().replace(/((http|https|ftp):\/\/[\w?=&.\/+-;#~%-]+(?![\w\s?&.\/;#~%"=-]*>))/g, '<a href="$1" target="_blank">$1</a> '))
+      $(this).html($(this).html().replace(/((http|https|ftp):\/\/[\w?=&./+-;#~%-]+(?![\w\s?&./;#~%"=-]*>))/g, '<a href="$1" target="_blank">$1</a> '))
```
Comment (HCrane) [inline assets/js/custom/Program.js:76]: Was this change intentionally done or via fixer?

**Review by HCrane — APPROVED** [review]

File: assets/js/base/Main.js (hunk 3 above)
Comment (dmetzner) [reply assets/js/base/Main.js:182]: Yes, but not directly.. It is converting `curX` to a boolean, then inverting it, then invert it again

image

But hey, never touch legacy code ;)

File: assets/js/custom/MyProfile.js (hunk 4 above)
Comment (dmetzner) [reply assets/js/custom/MyProfile.js:119]: by me, we are now using the newest version of sweetalert2

Comment (dmetzner) [reply assets/js/custom/MyProfile.js:119]: also, every const or let was introduced by hand ...
eslint just tells you that you should use it, but is not able to fix it automatically

File: assets/js/custom/Program.js (hunk 7 above)
Comment (dmetzner) [reply assets/js/custom/Program.js:76]: I did it intentionally since it throws a warning

File: assets/js/analytics/analytics.js (hunk 2 above)
Comment (dmetzner) [reply assets/js/analytics/analytics.js:1150]: 2 warnings less ;)
but not in the logic, no

**Pull request merged**